modules/bs.environment/data/bs.environment/function/celestial/get_sun_angle.mcfunction

```diff
@@ -0,0 +1,86 @@
+# ------------------------------------------------------------------------------------------------------------
+# Copyright (c) 2025 Gunivers
+#
+# This file is part of the Bookshelf project (https://github.com/mcbookshelf/bookshelf).
+#
+# This source code is subject to the terms of the Mozilla Public License, v. 2.0.
+# If a copy of the MPL was not distributed with this file, You can obtain one at http://mozilla.org/MPL/2.0/.
+#
+# Conditions:
+# - You may use this file in compliance with the MPL v2.0
+# - Any modifications must be documented and disclosed under the same license
+#
+# For more details, refer to the MPL v2.0.
+# ------------------------------------------------------------------------------------------------------------
+
+# A big thanks to @majoras16 and @Draco18s on Minecraft Forge (https://forums.minecraftforge.net/topic/29244-1710-detect-if-player-is-looking-at-the-sun/) for giving me pointers on how to implement this.
+
+# --- Original algorithm, from the Minecraft source code ---
+
+# double d = MathHelper.fractionalPart((double)this.fixedTime.orElse(time) / 24000.0 - 0.25);
+# double e = 0.5 - Math.cos(d * Math.PI) / 2.0;
+# float r = (float)(d * 2.0 + e) / 3.0f;
+# r = r * ((float) Math.PI * 2);
+
+# Then, to convert to degrees:
+
+# float result = (r * 180 / Math.PI) % 180) - 90;
+
+# The following code is the implementation of the above algorithm, adapted to the Minecraft scoreboard system.
+# Since we cannot use floatting point numbers, we use integers and scale them by 1000 to keep precision.
+
+# ----------------------------------------------------------
+
+# Get initial time values
+execute store result score #d bs.ctx run time query day
+execute store result score #t bs.ctx run time query daytime
+
+# --- Block 1: Calculate d (scale 1000) ---
+scoreboard players operation #d bs.ctx *= 24000 bs.const
+scoreboard players operation #d bs.ctx += #t bs.ctx
+scoreboard players operation #d bs.ctx *= 1000 bs.const
+scoreboard players operation #d bs.ctx /= 24000 bs.const
+scoreboard players operation #d bs.ctx -= 250 bs.const
+
+# --- Block 2: Fractional part of d ---
+# This is equivalent to fractionalPart(d) for a scale of 1000
+scoreboard players operation #d bs.ctx %= 1000 bs.const
+
+# --- Block 3: Calculate e ---
+scoreboard players operation #e bs.ctx = #d bs.ctx
+# In the original algorithm, d is multiplied by PI to obtain the angle in radians for the cosine function.
+# The cosine function from Bookshelf expects the angle in degrees, so we need to convert it to degrees instead of radians.
+scoreboard players operation #e bs.ctx *= 180 bs.const
+
+# Call cosine function with precision constraint
+# Prepare input: convert from scale 10³ to 10², since this is the precision of the Bookshelf's cosine function
+scoreboard players operation $math.cos.x bs.in = #e bs.ctx
+scoreboard players operation $math.cos.x bs.in /= 10 bs.const
+# Call the function
+function #bs.math:cos
```

Comment:
I feel like having a dep on math here is not worth it. We should inline the cos since it's a quite a simple function, also it's gonna be easier to stay on ctx and avoid copying scores around

Reply:
If you are okay with, I will just duplicate the function and change the input/output storage.

Reply:
I dont really care but i personally think it's better to inline it since we can certainly save a few commands this way. But this can be done in a future update

Reply:
cf what I've done.
I think this is similar to what you are talking about, but placed in a different file.

Reply:
Having it inline means one less function call, but if it's easier this way np

Reply:
Yeah, but also less clear, I think. I prefer to keep clarity, especially when we are talking about one function call.